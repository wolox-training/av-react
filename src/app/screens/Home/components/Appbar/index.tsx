import i18next from 'i18next';
import React from 'react';
import { useHistory } from 'react-router';

import { PATHS } from '~constants/paths';
import LocalStorageService from '~services/LocalStorageService';
import { TOKEN_KEY } from '~utils/constants';

import WoloxImg from '../../../Assets/wolox-logo.png';

import styles from './styles.module.scss';

export default function Appbar() {
  const history = useHistory();
  const logout = () => {
    LocalStorageService.removeValue(TOKEN_KEY);
    history.push(PATHS.login);
  };

  return (
    <header className={`${styles.appbar} row space-around middle`}>
      <img src={WoloxImg} alt={i18next.t('Home:logoAlt') as string} className={styles.logo} />
      <h2 className={styles.logout} onClick={logout}>
        {i18next.t('Home:logout')}
      </h2>
    </header>
  );
}
