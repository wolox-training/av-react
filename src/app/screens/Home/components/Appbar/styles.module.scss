@import 'variables/colors';

.appbar {
  border-top: 6px solid $cerulean;
  box-shadow: 0 0 12px $black;
  height: 70px;
  width: 100%;

  .logo {
    height: 70px;
    width: 212px;
  }

  .logout {
    color: $black;
    font-size: 18px;
    height: 18px;
    line-height: 21px;
    cursor: pointer;
    width: 63px;
  }
}
