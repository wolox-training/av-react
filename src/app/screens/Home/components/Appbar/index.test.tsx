import React from 'react';
import { screen, render } from '@testing-library/react';

import Appbar from './';

describe('Appbar test', () => {
  test('Should match snapshot', () => {
    const { container } = render(<Appbar />);
    expect(container).toMatchSnapshot();
  });

  test('Should render content', () => {
    render(<Appbar />);
    expect(screen.getByAltText('Home:logoAlt')).toBeInTheDocument();
    expect(screen.getByText('Home:logout')).toBeInTheDocument();
  });
});
