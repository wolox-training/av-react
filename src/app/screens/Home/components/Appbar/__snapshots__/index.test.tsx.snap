// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar test Should match snapshot 1`] = `
<div>
  <header
    class="appbar row space-around middle"
  >
    <img
      alt="Home:logoAlt"
      class="logo"
      src="wolox-logo.png"
    />
    <h2
      class="logout"
    >
      Home:logout
    </h2>
  </header>
</div>
`;
